docs/reference-capabilities/default-capabilities.md

```diff
@@ -0,0 +1,233 @@
+| recover (mutable)       | `iso`              | An immutable value (box or val) is recovered to `val` by default        |
+| literal (object)        | `ref`              | object literals default to same capabilities as their non-literal types |
+| literal (actor)         | `tag`              | object literals default to same capabilities as their non-literal types |
+| literal (primitive)     | `val`              | object literals default to same capabilities as their non-literal types |
+| lambda (no captures)    | `val`              | lambda capabilities depend on whether they capture their environment    |
+| lambda (captures)       | `ref`              | lambda capabilities depend on whether they capture their environment    |
+
+## Primitive default capabilities
+
+Let's start with primitives because they're easy. All primitives have `val` capabilities. They are immutable, they are read only, there is only one of each primitive, and they are accessible from all actors. So `val`.
+
+In fact, defining a primitive as `primitive val MyPrimitive` is a syntax error. Primitives are so `val` that you can't even tell them they are `val`.
+
+## Actor default capabilities
+
+Similar to primitives, actors always have a `tag` default capability. Like primitives, it is not possible to specify a different default capability for tags.
+
+```pony
+actor Actor
+```
+
+has `tag` capabilities
+
+```pony
+actor ref Actor
+```
+
+is illegal
+
+## Class default capabilities
+
+By default, a class has `ref` capability. Therefore, these two class definitions are equivalent:
+
+```pony
+class ref RefClass
+    ...
+
+class RefClass
+    ...
+```
+
+But whether specified by default or explicitly as `class ref` (or `class val` or any other capability) what does it actually mean? A class or actor's default capability only applies when you _define_ a field or local type. This can happen either as `let` or `var`, or when specifying a parameter on a function or behaviour.
+
+## Field, Variable, and Parameter capabilities
+
+For example, `String` is defined in the standard library as `class val String`. So it's default capability is the immutable shareable `val` capability. That means that all the `String`s in the following actor definition could be replaced with `String val`:
+
+```pony
+actor UsesStrings
+    let s1: String
+    var s2: String
+
+    new create(s1': String, s2': String) =>
+        s1 = s1'
+        s2 = s2'
+
+    be do_thing(s3: String) =>
+        None
+
+```
+
+However `StringBytes` is defined as `class ref`, so all the `StringBytes`' in the following class definition are equivalent to `StringBytes ref`:
+
+```pony
+class UsesStringBytes
+    let s1: StringBytes
+    var s2: StringBytes
+
+    new create(s1': StringBytes, s2': StringBytes) =>
+        s1 = s1'
+        s2 = s2'
+
+    fun do_thing(s3: StringBytes) =>
+        None
+```
+
+Notably, however, the default capability on a class **does not** influence the constructors for that class.
```

Comment:
What do you mean here?

Reply:
Do you mean the ability to override the default on a given constructor?

Reply:
So I think there's a misunderstanding here on what a "default capability on a class" is.

It's not a "default on the class" it is the default capability for the type if the type appears elsewhere in the source without a capability.

So...

`class val String` means that if I say: `let x: String` it means `let x: String val`.

Which has nothing to do with the capability of the instance created when calling a given constructor.

Reply:
Yeah, that was the confusion I was trying to resolve here. I originally expected there to be some correlation between the return type on the constructor and spent quite a bit of trial and error before it clicked that the constructor return type has nothing to do with the default on the class. It's one of those things that is too obvious to an experienced Pony developer and a new user needs to be told only once. Possibly it should go at the point class default reference capabilities are first introduced instead of here.

Reply:
Looking at this again, I realized that the chief purpose of this sentence was actually as a transition to the next section. I massaged it a bit, but it might still be a bit stiff.